### PR TITLE
Add FALLDAMAGE flag and add property to properly apply falling damage…

### PR DESCRIPTION
**Pull request by Cacodemon345**
… to the monsters

### CODE REVIEW

File: src/gamedata/g_mapinfo.cpp [hunk 1]
```diff
@@ -1644,6 +1644,7 @@ MapFlagHandlers[] =
 	{ "nolightfade",					MITYPE_SETFLAG3,	LEVEL3_NOLIGHTFADE, 0 },
 	{ "nocoloredspritelighting",		MITYPE_SETFLAG3,	LEVEL3_NOCOLOREDSPRITELIGHTING, 0 },
 	{ "forceworldpanning",				MITYPE_SETFLAG3,	LEVEL3_FORCEWORLDPANNING, 0 },
+	{ "propermonsterdamage",			MITYPE_SETFLAG3,	LEVEL3_PROPERMONSTERDAMAGE, 0 },
```
Comment (madame-rachelle) [inline src/gamedata/g_mapinfo.cpp:1647]: I don't like the name given here. Shouldn't it be "monsterfallingdamage" or something like that?

File: src/gamedata/g_mapinfo.h [hunk 2]
```diff
@@ -249,6 +249,7 @@ enum ELevelFlags : unsigned int
 	LEVEL3_EXITSECRETUSED		= 0x00000040,
 	LEVEL3_FORCEWORLDPANNING	= 0x00000080,	// Forces the world panning flag for all textures, even those without it explicitly set.
 	LEVEL3_HIDEAUTHORNAME		= 0x00000100,
+	LEVEL3_PROPERMONSTERDAMAGE	= 0x00000200,	// Properly apply falling damage to the monsters
```
Comment (madame-rachelle) [inline src/gamedata/g_mapinfo.h:252]: Let's fix the internal names too, while we're at it.

**Comment by Cacodemon345** [comment]
I named them as-is as they will make sure the falling damage is properly applied to monsters instead of always killing them.

**Comment by madame-rachelle** [comment]
Hmm... good point. Calling it "propermonsterfallingdamage" just makes the keyword way too long.

**Comment by Cacodemon345** [comment]
Well, should calling it "propmonsfalldamage" suffice?

**Comment by drfrag666** [comment]
What about "propermonsterfalldmg"?

**Comment by MekBoss** [comment]
"actorsfalldmg"

**Comment by MajorCooke** [comment]
Please fix the conflicts, so this can be added to QZDoom for testing.

**Comment by drfrag666** [comment]
Why not LEVEL3_PROPERMONSTERFALLDMG? LEVEL_FALLDMG_ZD already exists. Rachael said the keyword was too long.

**Comment by madame-rachelle** [comment]
That really was just me nitpicking. If Graf is okay with it, then I'm fine with it. I'll leave it to him to decide.

**Comment by coelckers** [comment]
If the conflicts are fixed this can be merged.

**Comment by Cacodemon345** [comment]
Can't say this PR has conflicts right now.

**Review by alexey-lysiuk — COMMENTED** [review]
> **This branch cannot be rebased due to conflicts**
> Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

However, squash and merge is available. It’s a viable option as commit history has zero value here.

**Pull request merged**